infrastructure/repositories/contract_repository.py

```diff
@@ -0,0 +1,14 @@
+from sqlalchemy.exc import SQLAlchemyError
+from infrastructure.repositories.base_repository import BaseRepository
+from infrastructure.models import Contract
+
+
+class ContractRepository(BaseRepository):
+    def get_contract(self, contract_name):
+        try:
+            contract = self.session.query(Contract).filter(Contract.contract_name == contract_name).first()
+            self.session.commit()
+        except SQLAlchemyError as e:
+            self.session.rollback()
+            raise e
+        return contract
```

Comment:
let's not return the DB object, make sure you convert DB to python object like define the domain and take which value from that ?

Reply:
Will make this change once we release publisher service changes.